controllers/grid/plugins/PluginGalleryGridHandler.inc.php

```diff
@@ -256,7 +256,15 @@ function upgradePlugin($args, $request) {
 	 * @param $isUpgrade boolean
 	 */
 	function installPlugin($args, $request, $isUpgrade = false) {
-		$redirectUrl = $request->getDispatcher()->url($request, ROUTE_PAGE, null, 'management', 'settings', array('website'), array('r' => uniqid()), 'plugins');
+		if ($request->getContext() == '') {
```

Comment:
`$request->getContext()` will return a `null` if you're in the administration area -- the empty string cast/compare is a little confusing. And I think it's clearer to read if the URLs are built for each area independently, rather than using `$handler` / `$path`. Add in `[]` for array syntax, and I'd recommend...

```php
if ($request->getContext()) {
    $redirectUrl = $request->getDispatcher()->url($request, ROUTE_PAGE, null, 'management, 'settings', ['website'], array('r' => uniqid()), 'plugins');
} else {
    $redirectUrl = $request->getDispatcher()->url($request, ROUTE_PAGE, null, 'admin', 'settings', null, array('r' => uniqid()), 'plugins');
}
```